### PR TITLE
test: add cypress setup

### PR DESCRIPTION
**Pull request by charmi-v**
#### Cypress Integration  

- **Dependency Added**: Included Cypress as a project dependency for end-to-end testing.  
- **Configuration Updates**: Set up essential configurations to enable seamless testing workflows and execution of Cypress test files.  
- **Initial Test File**: Added a basic test file to serve as a starting point for further test case development.  

## Why

To implement Cypress for automated E2E testing across modules, starting with primary features, to ensure consistent workflow validation, enable early bug detection, and enhance overall release reliability.

### Acceptance Criteria

- [x]  Add Cypress as a project dependency to enable E2E testing capabilities

## Issue

#1296 

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [x] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by charmi-v** [comment]
@evegufy  Should I proceed with updating the DEPENDENCIES file for the Cypress package in this PR, based on the IP check issue created and shared by @oyo  ?

**Review by evegufy — COMMENTED** [review]
Hi @charmi-v it looks really good!

I followed your documentation and was able to get until sharedidp login with the home e2e test case but then I encounter the following error:

image

Do you have an idea how to solve this?

**Comment by evegufy** [comment]
> @evegufy After merging the main branch, the generated artifact file is empty, causing the PR checks above to fail. Please take a look at your convenience.

The ClearlyDefined service was down. I just re-run the job and it ran successfully.

**Comment by charmi-v** [comment]
> Hi @charmi-v it looks really good!
> 
> I followed your documentation and was able to get until sharedidp login with the home e2e test case but then I encounter the following error:
> 
> image
> 
> Do you have an idea how to solve this?

Yes, @evegufy, this can be resolved with some changes to the test case. I will update the PR soon.

**Comment by charmi-v** [comment]
@evegufy , the code has been updated.

**Comment by SanaSalem01** [comment]
@charmi-v 

**Suggested Improvements for Reusability and Maintainability**
To enhance the reusability and maintainability of the framework, consider implementing the following changes:

**Proposed Changes**

- **Include the Company Name in cypress.config.ts**
Add a configuration field for the company name within the cypress.config.ts file to centralize its definition and allow dynamic updates.

**Test Environment Configuration Enhancements**

- Define test environment variables in a GitHub Actions YAML file for dynamic assignment at runtime.
- Use the cypress.config.ts file as a fallback to define default environment values.
- Set up a dynamic environment variable, such as ENVIRONMENT: ${{ inputs.environment }}, to specify the test environment during runtime.

**Benefits of These Changes**

- Improved Reusability
- Enhanced Maintainability
- Readability and Consistency
- Scalability
- Elimination of Hardcoding

**Comment by charmi-v** [comment]
@SanaSalem01  Agreed. The code has been updated to include the proposed change of adding the company name within the Cypress config file."

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+// @ref https://on.cypress.io/custom-commands
+Cypress.Commands.add('login', (usertype) => {
+  // creating a session for user login
+  cy.session(
+    [usertype],
+    () => {
+      const companyName = Cypress.env('company').name
+      cy.visit(Cypress.env('baseUrl'))
+      // Perform login on Keycloak login page
+      cy.origin(Cypress.env('keycloak').centralUrl, () => {
```
Comment (evegufy) [inline cypress/support/commands.ts:41]: Without this change, I always had an error that the Company is undefined
```suggestion
      cy.origin(Cypress.env('keycloak').centralUrl, { args: { companyName } }, ({ companyName }) => {
```

**Review by evegufy — CHANGES_REQUESTED** [review]
I just tested again and after getting over the company name undefined issue, I unfortunately still face the same issue as raised before: https://github.com/eclipse-tractusx/portal-frontend/pull/1357#pullrequestreview-2476216833

File: cypress/support/commands.ts (hunk 1 above)
Comment (charmi-v) [reply cypress/support/commands.ts]: Okay.. Got it.. updated the code

**Comment by evegufy** [comment]
Hi @charmi-v @manojava-gk I noticed that [the issue I encounter appears](https://github.com/eclipse-tractusx/portal-frontend/pull/1357#pullrequestreview-2476216833) only in the chrome browser and it might be related to policy settings https://docs.cypress.io/app/references/error-messages#Cypress-detected-policy-settings-on-your-computer-that-may-cause-issues

**Comment by SanaSalem01** [comment]
Hi @charmi-v  @evegufy 

I ran the test, and it works on both Chrome and Firefox without any issues. 

Everything looks good to me. :)👍

Screenshot_1061

Screenshot_1063

**Review by evegufy — APPROVED** [review]

**Pull request merged**